client/trino-client/src/main/java/io/trino/client/uri/ConnectionProperties.java

```diff
@@ -211,6 +230,27 @@ public static Map<String, ClientSelectedRole> parseRoles(String roles)
                     .collect(toImmutableMap(Map.Entry::getKey, entry -> mapToClientSelectedRole(entry.getValue())));
         }
 
+        public static String rolesToString(Map<String, ClientSelectedRole> roles)
+        {
+            return roles.entrySet().stream()
+                    .map(entry -> entry.getKey() + ":" + roleToString(entry.getValue()))
+                    .collect(Collectors.joining(";"));
+        }
+
+        private static String roleToString(ClientSelectedRole value)
```

Comment:
Previously role parsing was defunct/broken. Now it can be passed through the JDBC url